openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -1206,3 +1212,170 @@ def validate_glusterfsd_memory_usage(self, size_limit):
+                ('file', 'kubernetes.io/glusterfs'),
+                ('arbiter', 'kubernetes.io/glusterfs'),
+                ('block', 'gluster.org/glusterblock')):
+            secret_name = self.create_secret(
+                secret_name_prefix='auto-scale-secret-{}'.format(pvc_type),
+                secret_type=secret_type, skip_cleanup=skip_cleanup)
+            secrets.append(secret_name)
+
+        secret_file, secret_arbiter, secret_block = secrets
+
+        # Create Storage Classes
+        storage_classes = []
+        for sc, secret, pvc_type, isarbiter, pvc_count in (
+                ('file', secret_file, 'file', False, file_pvc_count),
+                ('file', secret_arbiter, 'arbiter', True, arbiter_pvc_count),
+                ('block', secret_block, 'block', False, block_pvc_count)):
+            if pvc_count:
+                sc_name = self.create_storage_class(
+                    sc_type=sc, secret_name=secret,
+                    sc_name_prefix='auto-scale-sc-{}'.format(pvc_type),
+                    vol_name_prefix='scale', is_arbiter_vol=isarbiter,
+                    skip_cleanup=skip_cleanup)
+                storage_classes.append(sc_name)
+
+        sc_file, sc_arbiter, sc_block = storage_classes
+
+        # Create PVC's
+        pvcs = []
+        for sc, pvc_count, prefix in (
+                (sc_file, file_pvc_count, "auto-scale-pvc-file"),
+                (sc_arbiter, arbiter_pvc_count, "auto-scale-pvc-arbiter"),
+                (sc_block, block_pvc_count, "auto-scale-pvc-block")):
+            pvclist = self.create_pvcs_in_batch(
+                sc, pvc_count,
+                pvc_name_prefix="{}-{}".format(prefix, name_prefix),
+                skip_cleanup=skip_cleanup)
+            pvcs.append(pvclist)
+
+        pvcs_file, pvcs_arbiter, pvcs_block = pvcs
+
+        # Create app pods
+        for pvc_list, pod_count, prefix in (
+                (pvcs_file, file_pod_count, "file"),
+                (pvcs_arbiter, arbiter_pod_count, "arbiter"),
+                (pvcs_block, block_pod_count, "block")):
+            self.create_app_pods_in_batch(
+                pvc_list, pod_count, dc_name_prefix=(
+                    "auto-scale-dc-{}-{}".format(prefix, name_prefix)),
+                skip_cleanup=skip_cleanup)
+
+    def verify_setup(self, prefix):
+        # Verify all bricks are up
+        self.check_vol_status()
+
+        # Wait for heal
+        wait_to_heal_complete(timeout=300, wait_step=10)
+
+        # Check mismatch of vols
+        match = (
+            'auto-scale-pvc-file-{}'.format(prefix),
+            'auto-scale-pvc-arbiter-{}'.format(prefix),
+            'auto-scale-pvc-block-{}'.format(prefix))
+
+        h_vol_list = heketi_volume_list(
+            self.heketi_client_node, self.heketi_server_url)
+        h_vol_list = [
+            vol.split('Name:')[1] for vol in h_vol_list.split('\n')
+            if any(m in vol for m in match)]
+
```

Comment:
There is already a library to match_heketi_and_gluster_volumes_by_prefix

Reply:
This library only match heketi volumes with glusster volumes with same code. Prefer to keep this code.